Stack/Opc.Ua.Core/Types/Encoders/JsonEncoder.cs

```diff
@@ -217,6 +217,10 @@ public string CloseAndReturnText()
             Close();
             if (m_destination == null)
             {
+                if (m_writer?.BaseStream is MemoryStream memoryStream)
+                {
+                    return Encoding.UTF8.GetString(memoryStream.ToArray());
+                }
                 return String.Empty;
```

Comment:
Maybe also add fallback to work with any stream?

else if (m_writer?.BaseStream != null)
{
      return Encoding.UTF8.GetString(m_writer.BaseStream.ToArray());
}
return string.Empty;

Reply:
Basestream is just a Stream, ToArray is not working directly. I'd not add too much of a convenience here and rather let the user handle it himself if its not a MemoryStream.